app/scripts/controllers/projects.js

```diff
@@ -16,13 +16,37 @@ angular.module('openshiftConsole')
     $scope.showGetStarted = false;
     $scope.canCreate = undefined;
 
+    var sortProjects = function() {
+      var sortID = _.get($scope, 'sortConfig.currentField.id', 'metadata.name');
+
+      $scope.projects = _.sortByOrder($scope.projects, function(project) {
+        return _.get(project, sortID).toLowerCase() || _.get(project, 'metadata.name').toLowerCase();
+      }, [$scope.sortConfig.isAscending ? 'asc' : 'desc']);
+    };
+
+    // Set up the sort configuration for `pf-sort`.
+    $scope.sortConfig = {
+      fields: [{
+        id: 'metadata.name',
+        title: 'Name',
+        sortType: 'alpha'
+      }, {
+        id: 'metadata.annotations["openshift.io/display-name"]',
+        title: 'Display Name',
+        sortType: 'alpha'
+      }],
+      isAscending: true,
+      onSortChange: sortProjects
+    };
+
     AlertMessageService.getAlerts().forEach(function(alert) {
       $scope.alerts[alert.name] = alert.data;
     });
     AlertMessageService.clearAlerts();
 
     AuthService.withUser().then(function() {
       watches.push(DataService.watch("projects", $scope, function(projects) {
+        $scope.emptyMessage = "No projects to show";
```

Comment:
If this only happens due to a filter, I'd prefer "The current filter is hiding all projects." Ideally with a "Clear filter" link.